packages/cspell-lib/src/lib/Settings/RegExpPatterns.test.ts

```diff
@@ -293,18 +293,19 @@ describe('Validate InDocSettings', () => {
         expect(r?.[0]).toEqual(expected);
     });
     test.each`
-        str                                      | expected
-        ${''}                                    | ${undefined}
-        ${'hello'}                               | ${undefined}
-        ${'commit 0x60975ea j'}                  | ${'0x60975ea'}
-        ${'only letters 0xfeed j'}               | ${'0xfeed'}
-        ${'commit 0xfeeeeed1 j'}                 | ${'0xfeeeeed1'}
-        ${'small value 0xf'}                     | ${'0xf'}
-        ${'trailing _ messes stuff up 0xf_ '}    | ${undefined}
-        ${'leading _ messes stuff up _0xf '}     | ${undefined}
-        ${'leading digit does not match 10xf '}  | ${undefined}
-        ${'leading letter does not match a0xf '} | ${undefined}
-        ${'commit c00ffee 0x0baad j'}            | ${'0x0baad'}
+        str                                          | expected
+        ${''}                                        | ${undefined}
+        ${'hello'}                                   | ${undefined}
+        ${'commit 0x60975ea j'}                      | ${'0x60975ea'}
+        ${'only letters 0xfeed j'}                   | ${'0xfeed'}
+        ${'commit 0xfeeeeed1 j'}                     | ${'0xfeeeeed1'}
+        ${'small value 0xf'}                         | ${'0xf'}
+        ${'trailing _ messes stuff up 0xf_ '}        | ${'0xf_'}
+        ${'trailing _ messes stuff up 0xbadc_0ffe '} | ${'0xbadc_0ffe'}
```

Comment:
@ccoVeille,

Here is the test.

Before, it wouldn't detect the `_` and would spell check `0xbadc`. Now it ignores things like `0xffff_eeee`.

Reply:
Oh I missed this test. I'm so used to the classic text files available in the cspell-dicts repository.

I've never faced this notation with hexa decimal

Thanks

Reply:
Opened this as a follow-up

- https://github.com/streetsidesoftware/cspell/issues/5942